src/lib/storage/segment_access_counter.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <array>
+#include <atomic>
+#include <map>
+#include <memory>
+#include <string>
+
+#include "storage/pos_list.hpp"
+#include "types.hpp"
+
+namespace opossum {
+class Table;
+
+// The SegmentAccessCounter is a collection of counters to register how often and how a segment is accessed.
+class SegmentAccessCounter {
+  friend class SegmentAccessCounterTest;
+
+ public:
+  using CounterType = std::atomic_uint64_t;
+
+  enum class AccessType {
+    Point /* Single point access */,
+    Sequential /* 0, 1, 1, 2, 3, 4 */,
+    Increasing /* 0, 0, 1, 2, 4, 8, 17 */ /* or decreasing */,
+    Random /* 0, 1, 0, 42 */,
+    Dictionary,
+    Count /* Dummy entry to describe the number of elements in this enum class. */
+  };
+
+  inline static const std::array<std::string, (size_t)AccessType::Count> access_type_string_mapping = {
+    "Point", "Sequential", "Increasing", "Random", "Dictionary"
+  };
```

Comment:
What is this used for? If this is just used for debugging, overloading `operator<<` makes the usage easier. If you actually need the string, it might be better to use an unordered_map here. It is slightly more expensive, but you don't run into issues when enum values are changed.

Reply:
I use the mapping to export the counters to a csv file. I could just create the headers in my export function. However, I'd perfer having the mapping at one central place to reduce the effort if somebody would ever change the access types. I'll create a map.